superset-frontend/src/dashboard/components/Header/index.jsx

```diff
@@ -365,18 +375,25 @@ class Header extends React.PureComponent {
     this.setState({ showingReportModal: false });
   }
 
+  handleReportModalclick() {
+    const attachedReportExists = this.props.report?.count > 0;
+    if (!attachedReportExists) {
+      this.showReportModal();
+    }
+  }
+
   canAddReportsModal() {
     if (!this.props.user) {
       // this is in the case that there is an anonymous user.
       return false;
     }
-    const roles = Object.keys(this.props.user?.roles);
+    const roles = Object.keys(this.props.user.roles);
```

Comment:
I would destructure out user, since you're accessing this.props.user a lot in this function

Reply:
also, since you're calling `map` on roles below, you're going to want to default roles to [] if null.